cmd/snap-confine/seccomp-support.c

```diff
@@ -152,24 +260,25 @@ bool sc_apply_seccomp_profile_for_security_tag(const char *security_tag)
 	// set on the system.
 	validate_bpfpath_is_safe(profile_path);
 
-	/* The extra space has dual purpose. First of all, it is required to detect
-	 * feof() while still being able to correctly read MAX_BPF_SIZE bytes of
-	 * seccomp profile.  In addition, because we treat the profile as a
-	 * quasi-string and use sc_streq(), to compare it. The extra space is used
-	 * as a way to ensure the result is a terminated string (though in practice
-	 * it can contain embedded NULs any earlier position). Note that
-	 * sc_read_seccomp_filter knows about the extra space and ensures that the
-	 * buffer is never empty. */
-	char bpf[MAX_BPF_SIZE + 1] = { 0 };
-	size_t num_read = sc_read_seccomp_filter(profile_path, bpf, sizeof bpf);
-	if (sc_streq(bpf, "@unrestricted\n")) {
+	//  when we stop supporting 14.04 we could just use hdr = {0}
+	struct sc_seccomp_file_header hdr;
+	memset(&hdr, 0, 1);
```

Comment:
I think you meant
```suggestion
	memset(&hdr, 0, sizeof hdr);
```

The `= { 0 }` syntax is a shorthand for initialising all the fields to zero.

Reply:
Done in 02c8a9ff81a8f036714eb4c0afce940cfa7c3c01